src/test/kotlin/com/example/passmanager/repositories/PassRepositoryImplTest.kt

```diff
@@ -11,207 +10,240 @@ import org.assertj.core.api.Assertions.assertThat
+        val passById = passRepository.findById(insertedPass!!.id.toString())
 
         // THEN
-        assertThat(passById?.id).isNotNull().isEqualTo(inserted.id)
+        passById.test()
+            .assertNext {
+                assertThat(it).usingRecursiveComparison()
+                    .ignoringFields(MongoPass::purchasedAt.name)
+                    .isEqualTo(insertedPass)
+            }
+            .verifyComplete()
     }
 
     @Test
     fun `inserting pass in collection should return created pass`() {
-        // WHEN
-        val inserted = passRepository.insert(passToCreate)
-        val insertedId = inserted.id
+        // GIVEN // WHEN
+        val insertedPass = mongoTemplate.insert(passToCreate)
 
         // THEN
-        assertTrue("pass must be persisted in db after creation") {
-            mongoTemplate.exists<MongoPass>(query(where("id").isEqualTo(insertedId)))
-        }
+        insertedPass
+            .test()
+            .assertNext { pass ->
+                mongoTemplate.findById<MongoPass>(pass!!)
+                    .doOnNext { passById -> assertThat(pass).isEqualTo(passById) }
+                    .subscribe()
+            }
+            .verifyComplete()
     }
 
     @Test
     fun `saving pass in collection show update existing pass by id`() {
         // GIVEN
-        val inserted = mongoTemplate.insert(passToCreate)
+        val insertedPass = mongoTemplate.insert(passToCreate)
         val changedPrice = BigDecimal.TEN
-        val updatedPass = inserted.copy(purchasedFor = changedPrice)
+        val updatedPass = insertedPass.map { it.copy(purchasedFor = changedPrice) }
 
         // WHEN
-        val saved = passRepository.save(updatedPass)
+        val saved = updatedPass.flatMap { passRepository.save(it) }
 
-        // THEN
-        assertThat(saved.purchasedFor).isEqualTo(changedPrice)
+        // WHEN
+        saved.test()
+            .assertNext { assertThat(it.purchasedFor).isEqualTo(changedPrice) }
+            .verifyComplete()
     }
 
     @Test
     fun `deleting pass by id should delete pass from collection`() {
         // GIVEN
-        val inserted = mongoTemplate.insert(passToCreate)
-        val insertedId = inserted.id
+        val insertedPass = mongoTemplate.insert(passToCreate).block()
+        val insertedPassId = insertedPass!!.id
 
         // WHEN
-        passRepository.deleteById(insertedId.toString())
+        val delete = passRepository.deleteById(insertedPassId.toString())
 
         // THEN
-        assertFalse("pass must not exist in db after deletion") {
-            mongoTemplate.exists<MongoPass>(query(where("id").isEqualTo(insertedId)))
-        }
+        delete.then(mongoTemplate.exists<MongoPass>(query(where(Fields.UNDERSCORE_ID).isEqualTo(insertedPassId))))
+            .test()
+            .assertNext { exists -> assertFalse("pass must not exist in collection after deletion") { exists } }
+            .verifyComplete()
```

Comment:
split into two parts
1. verify that passRepository.deleteById executes successfully
2. after execution verify that mongoTemplate.exists (here is acceptable to use .block())